chain-signatures/contract/src/lib.rs

```diff
@@ -533,15 +501,74 @@ impl VersionedMpcContract {
                     Ok(true)
                 } else {
                     Err(MpcContractError::VoteError(
-                        VoteError::UnexpectedProtocolState("resharing".to_string()),
+                        VoteError::UnexpectedProtocolState("Running".to_string()),
                     ))
                 }
             }
             _ => Err(MpcContractError::VoteError(
-                VoteError::UnexpectedProtocolState("resharing".to_string()),
+                VoteError::UnexpectedProtocolState("Running".to_string()),
             )),
         }
     }
+
+    /// Propose an update to the contract. [`Update`] are all the possible updates that can be proposed.
+    ///
+    /// returns Some(id) if the proposal was successful, None otherwise
+    #[handle_result]
+    pub fn propose_update(
+        &mut self,
+        code: Option<Vec<u8>>,
+        config: Option<Config>,
+    ) -> Result<UpdateId, VoteError> {
```

Comment:
I was asking about this pattern in one of my previous PRs. Does it make sense to return more specific errors on all, or all user-facing functions instead of the generic MPC error?

Reply:
yeah we should flatten the errors out similar to workspace errors because we have too many similar sounding errors that can pop up for categories of errors. Let me see if I can whip up something real quick for this before release